src/snowflake/cli/plugins/spcs/image_repository/manager.py

```diff
@@ -15,38 +21,43 @@ def get_schema(self):
     def get_role(self):
         return self._conn.role
 
-    def get_repository_url_list(self, repo_name: str) -> SnowflakeCursor:
-        role = self.get_role()
+    def get_repository_row(self, repo_name: str) -> Dict:
+        if not is_valid_identifier(repo_name):
+            raise ValueError(
+                f"repo_name '{repo_name}' is not a valid Snowflake identifier"
+            )
         database = self.get_database()
         schema = self.get_schema()
+        name_is_quoted = is_valid_quoted_identifier(repo_name)
 
-        registry_query = f"""
-            use role {role};
-            use database {database};
-            use schema {schema};
-            show image repositories like '{repo_name}';
-            """
-
-        return self._execute_query(registry_query)
-
-    def get_repository_url(self, repo_name):
-        database = self.get_database()
-        schema = self.get_schema()
+        # if repo_name is a quoted identifier, strip quotes before using as pattern
+        repo_name_raw = repo_name[1:-1] if name_is_quoted else repo_name.upper()
+        repository_list_query = (
+            f"show image repositories like '{escape_like_pattern(repo_name_raw)}'"
+        )
+        result_set = self._execute_query(repository_list_query, cursor_class=DictCursor)
+        results = result_set.fetchall()
 
-        result_set = self.get_repository_url_list(repo_name=repo_name)
+        # because SHOW LIKE uses case-insensitive matching, results may return multiple rows
+        results = [r for r in results if r["name"] == repo_name_raw]
 
-        results = result_set.fetchall()
         if len(results) == 0:
             raise ClickException(
                 f"Specified repository name {repo_name} not found in database {database} and schema {schema}"
             )
-        else:
-            if len(results) > 1:
-                raise Exception(
-                    f"Found more than one repositories with name {repo_name}. This is unexpected."
-                )
+        elif len(results) > 1:
+            raise Exception(
+                f"Found more than one repositories with name {repo_name}. This is unexpected."
+            )
+        return results[0]
```

Comment:
This pattern is common in Native Apps too -- suggest you take a look at `cli/plugins/nativeapp/manager.py` to see if there's anything you want to re-use

Reply:
Would you mind clarifying which pattern you are referring to? Do you mean the pattern of checking that there is exactly one result returned by a query or do you mean the error that is thrown? I looked through the native app manager and the utils but the closest thing I could find was the generic_sql_error_handler.

Reply:
The overall pattern of "I've got an identifier that I want to pipe through a `show ... like` construct and get the exact row that corresponds to my identifier". It feels like it should be a universal helper that everything calls.

Reply:
I've implemented a mixin that can be added to any class inheriting SqlExecutionMixin in `nativeapp/manager` let me know if that's what you had in mind.